test/token-service/test_tokenCreateTransaction.js

```diff
@@ -0,0 +1,8187 @@
+
+      // Check if this is expected to act like this
+      const convertedExpirationTimeToNanoSeconds =
+        expirationTime * 1_000_000_000;
+
+      expect(convertedExpirationTimeToNanoSeconds).to.equal(
+        Number(
+          await (
+            await mirrorNodeClient.getTokenData(tokenId)
+          ).expiry_timestamp,
+        ),
+      );
+    }
+
+    it("(#1) Creates a token with an expiration time of 0 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          expirationTime: 0,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#2) Creates a token with an expiration time of -1 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          expirationTime: -1,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#3) Creates a token with an expiration time of 9,223,372,036,854,775,807 (int64 max) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          expirationTime: 9223372036854775807n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#4) Creates a token with an expiration time of 9,223,372,036,854,775,806 (int64 max - 1) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          expirationTime: 9223372036854775806n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#5) Creates a token with an expiration time of 9,223,372,036,854,775,808 (int64 max + 1) seconds", async function () {
```

Comment:
same as https://github.com/hashgraph/hedera-sdk-tck/pull/252/files#r1794710905

Reply:
Aren't the test cases with `int64 MAX +1` and `int64 MIN` the same? 
I mean I am handling the overflow of the numbers in my TCK server, because i expect an `Integer` and not `BigInteger`

``` java
        Optional<Long> parsedExpirationTime;
        try {
            parsedExpirationTime = Optional.ofNullable((Long) jrpcParams.get("expirationTime")); // handle long boundary
        } catch (Exception e) {
            parsedExpirationTime = Optional.of(((BigInteger) jrpcParams.get("expirationTime")).longValue()); // will overflow to negative number
        }

```

I am still a big confused if we really need `int64 MAX +1` cases.

Reply:
I believe the idea here was just to make sure the SDKs could parse both representations of the same underlying bit structure properly. I don't feel strongly either way about keeping or removing, @jsync-swirlds you have an opinion?

Reply:
We need these "out of bounds" edge case tests so that the TCK ensures that all SDKs handle these values (which are _valid_ in some languages) correctly and consistently.

Protocol Buffers do not translate these to `BigInteger` for Java, they translate as twos-compliment integers (so `INT64_MAX + 1` is, indeed, bitwise identical to `INT64_MIN`), but they do not translate the same in other SDK languages.
It seems, to me, that anywhere that different SDKs are expected to _correctly_ interpret the same protocol buffer serialized bytes in different ways are worth testing.

Reply:
But how is this testing the SDKs specifically when, for example, I have a method `setValueX(long value)` in my SDK and we are trying to "test" this method by passing value outside of the `long` boundary. 

I don't really see the benefit of doing this, as it only introduces overhead in the TCK server to "overflow the value" artificially.

Also, as far as I see, these tests are working in the C++ SDK because It has most (if not all) of it's `int64` values as `uint64`, see `amount`in https://github.com/hashgraph/hedera-sdk-cpp/blob/main/src/sdk/main/src/CustomFixedFee.cc , `maxSupply` in https://github.com/hashgraph/hedera-sdk-cpp/blob/main/src/sdk/main/src/TokenCreateTransaction.cc and many more. 
SDKs like Go, for example, have the above mentioned values as `int64` , so how can we make the behaviour of the TCK server consistent? 

I haven't ran the C++ TCK server but I'm pretty sure that when we pass `INT64_MAX + 1` it maps correctly because the parameters are `uint64` and not `int64` (or `long` in other languages) and then it's getting static casted like this `static_cast<int64_t>(mMaxSupply)` in order to overflow.

@rwalworth @jsync-swirlds

Reply:
@jsync-swirlds what do you mean when you say "but they do not translate the same in other SDK languages."? Like an `int64` protobuf field wouldn't translate to a twos-complement signed integer type for another language? I think I agree with @0xivanov in that, at least for signed integer types, it seems a bit superfluous to test `INT64_MAX + 1` and `INT64_MIN` since they're the same underlying bit structure. Same with testing `UINT64_MAX` and `-1` for signed integer types. I can see the benefits of testing the `UINT64_MAX` values however if we have an unsigned protobuf value, solely because services doesn't support unsigned integers; I think that would be an overflow case that we'd want to test and make sure services handles properly.

Reply:
Other languages interpret some protobuf types differently than Java (e.g. some languages may interpret `uint64` as a string, "Bignum", or similar, rather than twos-compliment binary, some _do_ support unsigned integers, etc...).  This is most pronounced for the unsigned types, not so much for signed types.
As a result of these differences it makes sense to test the edges of _signed integers_ for `uint64` and `uint32` values, including int_max+1 and int_min so that we have an opportunity to catch these differences.

int64_max+1 and int64_min are interesting in that they _are_ identical when stored unsigned or twos-compliment, but differ if stored as a string or Bignum, for instance, and that is why I suggest testing those two specifically.

Note, _in my opinion_ no SDK should be casting unsigned values to signed values that are _different_ from the protocol buffer definition; that would be a very strange behavior.

All edge cases are, in my opinion, suggestions; for any given test those edges might or might not be relevant.
If the protobuf type is `int32` or `int64`, then these edges don't make as much sense, of course.

Reply:
@jsync-swirlds and I talked offline... for signed 64-bit types, we think testing something to the effect of `[0, -1, INT64_MAX - 1, INT64_MAX, INT64_MIN, INT64_MIN + 1]` as well as maybe one standard valid value should be sufficient. Does this sound good to you @0xivanov @ivaylogarnev-limechain?